File: android/src/main/kotlin/io/github/kokemus/cross_bluetooth_api/events/CharacteristicValueChangedEvent.kt
package io.github.kokemus.cross_bluetooth_api.events

class CharacteristicValueChangedEvent(
    private val deviceId: String,
    private val serviceUUID: String,
    private val characteristicUUID: String,
    private val value: ByteArray
) : Event(EventName.CHARACTERISTIC_VALUE_CHANGED) {
    override fun toMap(): Map<String, Any> {
        return super.toMap() + mapOf(
            "deviceId" to deviceId,
            "serviceUUID" to serviceUUID,
            "characteristicUUID" to characteristicUUID,
            "value" to value
        )
    }
}

File: ios/cross_bluetooth_api/Sources/cross_bluetooth_api/Events/CharacteristicValueChangedEvent.swift
import Foundation
import Flutter

public class CharacteristicValueChangedEvent: Event {
    let deviceId: String
    let serviceUUID: String
    let characteristicUUID: String
    let value: Data

    init(
        deviceId: String,
        serviceUUID: String,
        characteristicUUID: String,
        value: Data
    ) {
        self.deviceId = deviceId
        self.serviceUUID = serviceUUID
        self.characteristicUUID = characteristicUUID
        self.value = value
        super.init(name: .characteristicvaluechanged)
    }

    override func toMap() -> [String:AnyObject] {
        var map = super.toMap()
        map["deviceId"] = deviceId as AnyObject
        map["serviceUUID"] = serviceUUID as AnyObject
        map["characteristicUUID"] = characteristicUUID as AnyObject
        map["value"] = value as AnyObject
        return map
    }
}
